Bartalamey

Bartalameu opened the pull request.

Bartalameu committed:
1-Refactoirng code
2-add few styles for toolbar
3-make toolbar for activity_test

Bartalameu committed:
Merge remote-tracking branch 'origin/bartalamey' into bartalamey

Bartalameu committed:
Merge branch 'bartalamey' into masterDenis

Bartalameu committed:
update branches

Bartalameu committed:
Merge remote-tracking branch 'Denis/master' into bartalamey

# Conflicts:
#	app/src/main/res/values/strings.xml

Bartalameu committed:
Add new classes and xml  for code review
Change AndroidManifest for simplier  debuging

bboyzlodey commented on app/src/main/AndroidManifest.xml:
```diff
@@ -34,21 +34,32 @@
 			android:name=".activity.HomeActivity"
-			android:exported="true" />
+			android:exported="true" >
+			<intent-filter>
```
Это для тестов? Удали

bboyzlodey commented on app/src/main/res/values/colors.xml:
```diff
@@ -1,7 +1,7 @@
 <resources>
     <color name="colorPrimary">#6200EE</color>
-    <color name="colorPrimaryDark">#7E5EC5</color>
+    <color name="colorPrimaryDark">#3700B3</color>
```
Зачем ты поменял цвет?!)

Bartalameu replied:
честно не могу вспомнить, скорее всего забыл вернуть как было

bboyzlodey commented on app/src/main/res/values/strings.xml:
```diff
@@ -51,4 +51,11 @@
+    <string name="test_is_done">Тест пройден !</string>
+    <string name="image_next_card">image_next_card</string>
+    <string name="check_knowledge">Проверим знания!</string>
+    <string name="How_can_i_help">Как я могу помочь?</string>
```
Не надо переводить строку на английский и писать ее как name
Name - отображает суть, а не значение

bboyzlodey replied:
header_eco_article - как пример

Bartalameu replied:
писать гаммотные коммиты, название переменных и тд, для меня это отдельный вид исскуства, я пока в этом оч плох, но я понял

bboyzlodey replied:
Так мы для этого (и не только для этого) делаем код ревью)

bboyzlodey commented on app/src/main/res/layout/advice_view.xml:
```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Это не вью, а item для ресайклера
app/src/main/res/layout/advice_view.xml - это название макета для инфлейта кастомной вьюшки
item_advice - хорошо сюда подойдет

bboyzlodey commented on app/src/main/res/layout/activity_course_card.xml:
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.coordinatorlayout.widget.CoordinatorLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Зачем нужно было трогать этот файл?

Bartalameu replied:
опять же , забыл вернуть в исходное положение, прости

bboyzlodey commented on app/src/main/res/layout/activity_eco_cards.xml:
```diff
@@ -6,38 +6,51 @@
-    <androidx.core.widget.NestedScrollView
-        android:layout_width="match_parent"
-        android:layout_height="match_parent">
+            <com.google.android.material.appbar.AppBarLayout
```
Ты изменил файл и нигде его не используешь в своем коде. Зачем тогда это комитить?

bboyzlodey commented on app/src/main/res/layout/activity_eco_cards.xml:
```diff
@@ -6,38 +6,51 @@
+                        android:layout_marginRight="16dp"
+                        android:adjustViewBounds="true"
+                        android:background="@null"
+                        android:onClick="onBackBtn"
```
Так мы больше не пишем :)

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    fun submitList(card: List<Any>) {
+        cards.addAll(card)
```
Перед добавлением в лист, его надо очистить

bboyzlodey replied:
?

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+
+class TipsAdapter (context: Context? , onAdviceListener: OnAdviceListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private var cards = arrayListOf<Any>()
+    private val inflater: LayoutInflater = LayoutInflater.from(context)
```
Да, в моем примере написано так...
Но так делать нельзя
Не нужно передавать контекст в адаптер

Контекст пожно взять у вью или у парента

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+
```
val layoutInflater = LayoutInflater.from(parent.context)

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+            else -> ADVICE_VIEW
+        }
+    }
+         interface OnAdviceListener {
```
Интерфейс лишний

Bartalameu replied:
без интерфейса не знаю как сделать, если можно без то поищу

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+        val description: TextView = itemView.findViewById(R.id.card_why_shortDescription)
+        val card_title: TextView = itemView.findViewById(R.id.card_title)
+        val card_category: TextView = itemView.findViewById(R.id.card_category)
+         val clickListener = itemView.setOnClickListener(this )
```
Что будет храниться в переменной? Оно тебе как-то в дальнешем понадобится?
Для иницилизации есть блок `init {
}`

Bartalameu replied:
долбаный kotlin cэр

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+import skarlat.dev.ecoproject.includes.dataclass.EcoSoviet
+
+class TipsAdapter (context: Context? , onAdviceListener: OnAdviceListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private var cards = arrayListOf<Any>()
```
У нас карточки или типсы? Зачем нам изменяемая переменная? Мы ее изменяем как-то?

Bartalameu replied:
и карточки и типсы, я ничего не придумал умнее, я в CardActivityMine ложу EcoCard в cards = arrayListOf<Any>() и типсы , и дальше в адаптере вытягиваю из этого массива данные , счас нету описания краткого но потом же будет, и я так понял что  из EcoCard брать надо будет, и заполнять кастомное вью.
все на что способен к сожалению

bboyzlodey reviewed (commented):
Уже лучше!

В общем создание копий, или изменение файлов, которые никак не влияют на работу приложения - это ненужная процедура. Ты можешь у себя делать, что хочешь. Но в пул реквест отправлять не нужно

Если ты хочешь поменять название файла, то это должно быть осознано и оправдано

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,97 @@
+
+    private inner class AdviceViewHolder internal constructor(itemView: View, onAdviceListener: OnAdviceListener) : RecyclerView.ViewHolder(itemView) , View.OnClickListener  {
+        val description: TextView = itemView.findViewById(R.id.card_why_shortDescription)
+        val card_title: TextView = itemView.findViewById(R.id.card_title)
```
Имена переменных пишутся в верблжьем стиле. Т.е.:
cardTitle

Bartalameu replied:
нехочу хвастаться, но единственное мое личное достижение, это то что я вчера , еще до твоего ответа так исправил )))

Bartalameu committed:
1-Refactoring CardActivity.java to CardActivity.kt
2-Delete class CardActivityMine.java and activity_eco_cards_mine.xml
3-Make  setOnClickListener in  TipsAdapter.kt, for whyItPossible,  by invoke instead of interface
4-Refactoring name declarations

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/activity/CardActivity.kt:
```diff
@@ -0,0 +1,84 @@
+
+
+    private fun initiList() {
+        adapter.clearList()
```
Очистка листа должна просходить внутри адаптера. Адаптер должен самостоятельно очистить лист. Логику очищения листа добавь в тело метода `submitList`

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,100 @@
+class TipsAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private var cards = mutableListOf<Any>()
+
+    var onItemClick: ((EcoCard) -> Unit)? = null
```
Уже лучше
Но тут нам нужна не `var`, а `val`. Предпочтительнее использовать `val`

Ты можешь написать такой конструктор для адаптера: `class TipsAdapter( private val onOpenArticleClicked: ((EcoCard) -> Unit)) ...`

Bartalameu replied:
не понимаю как это работает, тяжело для меня такое

bboyzlodey replied:
https://kotlinlang.ru/docs/reference/classes.html - Заголовок контрукторы
https://kotlinlang.ru/docs/reference/basic-syntax.html - Опеределие внутренних переменных

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,100 @@
+    }
+
+     fun clearList () {
+      cards.clear()
```
Перенеси в тело мето метода `submitList`

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/adapter/TipsAdapter.kt:
```diff
@@ -0,0 +1,100 @@
+        val advice = cards[position] as EcoCard
+        holder.cardTitle.text = advice.title
+        holder.cardCategory.text = advice.description
+        holder.description.text = "Test text Test text !!!!!!"
```
?

Bartalameu replied:
не понял что не так

bboyzlodey commented on app/src/main/res/values/strings.xml:
```diff
@@ -51,4 +51,11 @@
     <string name="name_changed_success">Имя пользователя изменено!</string>
     <string name="mail_changed_success">Почта изменена!</string>
     <string name="count_users_format">%04d</string>
+    <string name="next_card_open">Следующая карточка открыта!</string>
```
Это зачем? У нас такого в ui

bboyzlodey commented on app/src/main/java/skarlat/dev/ecoproject/activity/CardActivity.kt:
```diff
@@ -0,0 +1,84 @@
+import java.io.IOException
+
+var TAG = "EcoCardActivity"
+lateinit var ecoCard: EcoCard
```
1) Модификатор доступа. Если поле не используется в другом классе ставь `private`
2) `lateinit var` - это очень плохая практика, которая приводит к вылетам. Например, при открытии карточки вылетает этот exceprion:
Эмулятор Pixel 2 API 29
`2021-01-30 21:16:47.407 9850-9850/skarlat.dev.ecoproject E/AndroidRuntime: FATAL EXCEPTION: main
    Process: skarlat.dev.ecoproject, PID: 9850
    java.lang.RuntimeException: Unable to start activity ComponentInfo{skarlat.dev.ecoproject/skarlat.dev.ecoproject.activity.CardActivity}: kotlin.UninitializedPropertyAccessException: lateinit property card has not been initialized
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3270)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: kotlin.UninitializedPropertyAccessException: lateinit property card has not been initialized
        at skarlat.dev.ecoproject.activity.CardActivityKt.getCard(CardActivity.kt:22)
        at skarlat.dev.ecoproject.activity.CardActivity.initiList(CardActivity.kt:57)
        at skarlat.dev.ecoproject.activity.CardActivity.onCreate(CardActivity.kt:44)
        at android.app.Activity.performCreate(Activity.java:7802)
        at android.app.Activity.performCreate(Activity.java:7791)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1299)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) `

bboyzlodey commented on .idea/shelf/Uncommitted_changes_before_Checkout_at_12_25_20_10_17_PM__Default_Changelist_.xml:
```diff
@@ -0,0 +1,4 @@
+<changelist name="Uncommitted_changes_before_Checkout_at_12_25_20_10_17_PM_[Default_Changelist]" date="1608931057410" recycled="false" toDelete="true">
+  <option name="PATH" value="$PROJECT_DIR$/.idea/shelf/Uncommitted_changes_before_Checkout_at_12_25_20_10_17_PM_[Default_Changelist]/shelved.patch" />
+  <option name="DESCRIPTION" value="Uncommitted changes before Checkout at 12/25/20 10:17 PM [Default Changelist]" />
```
Все, что выше не нужно включать в пул реквест!

bboyzlodey commented on app/src/main/res/layout/activity_eco_cards.xml:
```diff
@@ -6,161 +6,84 @@
+                    >
+                </androidx.recyclerview.widget.RecyclerView>
+
+                <Button
```
Кнопку тоже в ресайклер

bboyzlodey commented on app/src/main/res/layout/activity_eco_cards.xml:
```diff
@@ -6,161 +6,84 @@
+        android:layout_height="wrap_content"
+        android:layout_below="@+id/appBar">
+
+        <androidx.core.widget.NestedScrollView
```
NestedScrollView не должно быть. Идея как раз в том, что мы избавляется от него и переносим все в ресайклер.

bboyzlodey reviewed (commented):
1) Приложение крашится при открытии карточки
2) Некоторые старые замечания остались актуальными (пожалуйста повнимательнее)
3) Рекомендую смотреть уже мной написанные примеры и внимательно их изучать

На будущее: если хочешь что-то переписать на котлин из существующего, то сначала скажи мне. Я сделаю это сам, а ты сделаешь пулл (обновление) из моей ветки. Это для того, чтобы не было такого, что у тебя код на котлин, а у меня тот же активити на java

bboyzlodey reviewed (changes requested):
Пожалуйста, там где исправляешь, пометь, что исправлено

bboyzlodey commented:
isqua.ru › blog › 2017/04/12
Все результаты
Как открыть пулл-реквест в Github и не облажаться - isqua
